### ISSUE TITLE
After deleting the entity, the target entity still has incoming links.

### ISSUE THREAD

**Issue by Belzebbuz** [friflo/Friflo.Engine.ECS#13]
This test fail
```csharp
var store = new EntityStore();
var entityA = store.CreateEntity();
var entityB = store.CreateEntity();
entityA.AddRelation(new Link(entityB));
		
entityA.DeleteEntity();

var incomingLinks = entityB.GetIncomingLinks<Link>();
incomingLinks.Count.Should().Be(0);

private readonly struct Link(Entity target) : ILinkRelation
{
    public Entity GetRelationKey() => target;
}
```
![image](https://github.com/user-attachments/assets/21666bb5-603d-436c-84d9-248a019fa19d)

**Comment by friflo** [comment]
add test [`Test_Relations_GitHub_13`](https://github.com/friflo/Friflo.Engine.ECS/blob/main/src/Tests/ECS/Relations/Test_Relations_GitHub_13.cs)
to reproduce bug.

**Comment by friflo** [comment]
Fixed by [Fix: Remove incoming links if removing linked target entity](
https://github.com/friflo/Friflo.Engine.ECS/commit/54f0c1765af70704e0a66a6afe71ba4ed087fb77)

**Comment by friflo** [comment]
Fixed in 3.0.0-preview.14
